community/cypher/cypher/src/test/scala/org/neo4j/cypher/MatchAcceptanceTest.scala

```diff
@@ -21,11 +21,107 @@ package org.neo4j.cypher
 
 import org.neo4j.cypher.internal.PathImpl
 import org.neo4j.graphdb._
+import org.neo4j.graphdb.factory.GraphDatabaseFactory
 
 import scala.collection.JavaConverters._
 
 class MatchAcceptanceTest extends ExecutionEngineFunSuite with QueryStatisticsTestSupport with NewPlannerTestSupport {
 
+  test("a merge following a delete of multiple rows should not match on a deleted entity") {
```

Comment:
There were also MergeNodeAcceptanceTest and MergeRelationshipAcceptanceTest classes, but they also are not the right place. Easiest is to make a new MergeAcceptanceTest class, and then see if there is an easy way to port to compatibility-suite.